string data
bool shouldCheckForAgentID
uint32[] agentIDs